// pharos-agent-kit/dist/langchain/dexscreener/token_data_ticker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PharosTokenDataByTickerTool = void 0;
const tools_1 = require("langchain/tools");
class PharosTokenDataByTickerTool extends tools_1.Tool {
    constructor(pharosKit) {
        super();
        this.pharosKit = pharosKit;
        this.name = "pharos_token_data_by_ticker";
        this.description = `Get the token data for a given token ticker

  Inputs: ticker is required.
  ticker: string, eg "USDC" (required)`;
    }
    async _call(input) {
        try {
            const ticker = input.trim();
            const tokenData = await this.pharosKit.getTokenDataByTicker(ticker);
            return JSON.stringify({
                status: "success",
                tokenData,
            });
        }
        catch (error) {
            return JSON.stringify({
                status: "error",
                message: error.message,
                code: error.code || "UNKNOWN_ERROR",
            });
        }
    }
}
exports.PharosTokenDataByTickerTool = PharosTokenDataByTickerTool;
