// pharos-agent-kit/dist/actions/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ACTIONS = void 0;
const createImage_1 = __importDefault(require("./agent/createImage"));
const getWalletAddress_1 = __importDefault(require("./agent/getWalletAddress"));
const get_info_1 = __importDefault(require("./agent/get_info"));
const elfa_ai_actions_1 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_2 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_3 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_4 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_5 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_6 = require("./elfa_ai/elfa_ai_actions");
const elfa_ai_actions_7 = require("./elfa_ai/elfa_ai_actions");
const getCoingeckoLatestPools_1 = __importDefault(require("./coingecko/getCoingeckoLatestPools"));
const getCoingeckoTokenPriceData_1 = __importDefault(require("./coingecko/getCoingeckoTokenPriceData"));
const getCoingeckoTopGainers_1 = __importDefault(require("./coingecko/getCoingeckoTopGainers"));
const getCoingeckoTrendingPools_1 = __importDefault(require("./coingecko/getCoingeckoTrendingPools"));
const getCoingeckoTrendingTokens_1 = __importDefault(require("./coingecko/getCoingeckoTrendingTokens"));
const fetch_price_action_1 = __importDefault(require("./defillama/fetch_price_action"));
const get_protocol_tvl_action_1 = __importDefault(require("./defillama/get_protocol_tvl_action"));
const tokenDataByTicker_1 = __importDefault(require("./dexscreener/tokenDataByTicker"));
exports.ACTIONS = {
    GET_INFO_ACTION: get_info_1.default,
    WALLET_ADDRESS_ACTION: getWalletAddress_1.default,
    CREATE_IMAGE_ACTION: createImage_1.default,
    ELFA_PING_ACTION: elfa_ai_actions_1.elfaPingAction,
    ELFA_API_KEY_STATUS_ACTION: elfa_ai_actions_2.elfaApiKeyStatusAction,
    ELFA_GET_SMART_MENTIONS_ACTION: elfa_ai_actions_3.elfaGetSmartMentionsAction,
    ELFA_GET_TOP_MENTIONS_BY_TICKER_ACTION: elfa_ai_actions_4.elfaGetTopMentionsByTickerAction,
    ELFA_SEARCH_MENTIONS_BY_KEYWORDS_ACTION: elfa_ai_actions_5.elfaSearchMentionsByKeywordsAction,
    ELFA_TRENDING_TOKENS_ACTION: elfa_ai_actions_6.elfaTrendingTokensAction,
    ELFA_SMART_TWITTER_ACCOUNT_STATS_ACTION: elfa_ai_actions_7.elfaSmartTwitterAccountStats,
    GET_COINGECKO_LATEST_POOLS_ACTION: getCoingeckoLatestPools_1.default,
    GET_COINGECKO_TOKEN_PRICE_DATA_ACTION: getCoingeckoTokenPriceData_1.default,
    GET_COINGECKO_TOP_GAINERS_ACTION: getCoingeckoTopGainers_1.default,
    GET_COINGECKO_TRENDING_POOLS_ACTION: getCoingeckoTrendingPools_1.default,
    GET_COINGECKO_TRENDING_TOKENS_ACTION: getCoingeckoTrendingTokens_1.default,
    DEFILLAMA_FETCH_PRICE: fetch_price_action_1.default,
    DEFILLAMA_GET_PROTOCOL_TVL: get_protocol_tvl_action_1.default,
    GET_TOKEN_DATA_BY_TICKER: tokenDataByTicker_1.default,
};
