// pharos-agent-kit/dist/network/network.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.getChain = exports.NETWORK_ID_TO_VIEM_CHAIN = exports.NETWORK_ID_TO_CHAIN_ID = exports.CHAIN_ID_TO_NETWORK_ID = exports.pharosDevnet = void 0;
const chains = __importStar(require("viem/chains"));
/**
 * Pharos Chain Configuration
 */
exports.pharosDevnet = {
    id: 50002,
    name: 'Pharos Devnet',
    nativeCurrency: {
        name: 'Ethereum',
        symbol: 'ETH',
        decimals: 18,
    },
    rpcUrls: {
        default: {
            http: ['https://devnet.dplabs-internal.com'],
        },
        public: {
            http: ['https://devnet.dplabs-internal.com'],
        },
    },
    blockExplorers: {
        default: {
            name: 'PharosExplorer',
            url: 'https://pharosscan.xyz/',
            apiUrl: '',
        },
    },
};
exports.CHAIN_ID_TO_NETWORK_ID = {
    50002: "pharos-devnet",
};
exports.NETWORK_ID_TO_CHAIN_ID = Object.entries(exports.CHAIN_ID_TO_NETWORK_ID).reduce((acc, [chainId, networkId]) => {
    acc[networkId] = String(chainId);
    return acc;
}, {});
exports.NETWORK_ID_TO_VIEM_CHAIN = {
    "pharos-devnet": exports.pharosDevnet,
};
/**
 * Get a chain from the viem chains object or custom chains
 *
 * @param id - The chain ID
 * @returns The chain
 */
const getChain = (id) => {
    // Check custom chains first
    if (id === String(exports.pharosDevnet.id)) {
        return exports.pharosDevnet;
    }
    // Then check viem chains
    const chainList = Object.values(chains);
    return chainList.find((chain) => chain.id === parseInt(id));
};
exports.getChain = getChain;
